### PR TITLE
db-refactor-6

### PR DESCRIPTION
**Pull request by freesig**
- Final changes to cascade to work with space level databases.
- Adds the ability for cascade to get private entries for a given author.

- Depends on #1120 
- Followed by #1122

### CODE REVIEW

**Review by thedavidmeister — APPROVED** [review]

File: crates/holochain_cascade/src/lib.rs [hunk 1]
```diff
@@ -390,7 +454,12 @@ where
     ) -> CascadeResult<Option<EntryDetails>> {
         let authoring = self.am_i_authoring(&entry_hash.clone().into())?;
         let authority = self.am_i_an_authority(entry_hash.clone().into()).await?;
-        let query = GetEntryDetailsQuery::new(entry_hash.clone());
+        let query = match self.private_data.clone() {
+            Some(author) => {
+                GetEntryDetailsQuery::with_private_data_access(entry_hash.clone(), author)
+            }
+            None => GetEntryDetailsQuery::new(entry_hash.clone()),
+        };
```
Comment (neonphog) [inline crates/holochain_cascade/src/lib.rs:462]: this change seems to be duplicate-ly applied multiple times in this file. Is it worth a function abstraction?

**Review by neonphog — APPROVED** [review]
one question, otherwise lgtm

File: crates/holochain_cascade/src/lib.rs (hunk 1 above)
Comment (freesig) [reply crates/holochain_cascade/src/lib.rs:462]: Note to self this is fixed in commit `bb78fa7c3e66363ee839ca0339b5027735678d83`

Comment (freesig) [reply crates/holochain_cascade/src/lib.rs:462]: Resolved in https://github.com/holochain/holochain/commit/bb78fa7c3e66363ee839ca0339b5027735678d83

**Pull request closed without merging**